library/bignum_mod.c

```diff
@@ -39,7 +39,7 @@ int mbedtls_mpi_mod_residue_setup( mbedtls_mpi_mod_residue *r,
                                    mbedtls_mpi_uint *p,
```

Comment:
Preexisting, but please fix this since it's necessary to get the checks and their tests right (or make a precursor PR to fix and test this function): the documentation can't be right. For the parameter `r`, it states

> (In particular, it must have at least as many limbs as the modulus \p m.)

But `r` is evidently an output of the function, it has no input constraints. The constraint is actually on `p`.